sdk/storage/Azure.Storage.DataMovement/src/Shared/CheckpointerExtensions.cs

```diff
@@ -148,6 +152,16 @@ internal static void Write(this BinaryWriter writer, DateTimeOffset? value)
             writer.Write(value?.Offset.Ticks ?? 0L);
         }
 
+        /// <summary>
+        /// Writes two int zero values to represent the length and offset of a value
+        /// that is preserved.
+        /// </summary>
+        internal static void WriteEmptyLengthOffset(this BinaryWriter writer)
+        {
+            writer.Write(0);
+            writer.Write(0);
```

Comment:
I think we should write the offset as `-1` to indicate it's not present. I thought I had this pattern already somewhere? Maybe in the job part schema or something.

Reply:
done